File: specs/Qowaiv.CodeAnalysis.Specs/UseTestableTimeProvider_specs.cs
using CodeAnalysis.TestTools;
using NUnit.Framework;
using CS = Qowaiv.CodeAnalysis.CSharp;
using VB = Qowaiv.CodeAnalysis.VisualBasic;

namespace Rules.UseTestableTimeProvider_specs
{
    public class Verify
    {
        [Test]
        public void CSharp()
            => new CS.UseTestableTimeProvider()
            .ForCS()
            .AddSource(@"Cases\UseTestableTimeProvider.cs")
            .Verify();

        [Test]
        public void VisualBasic()
            => new VB.UseTestableTimeProvider()
            .ForVB()
            .AddSource(@"Cases\UseTestableTimeProvider.vb")
            .Verify();
    }
}

File: specs/Qowaiv.CodeAnalysis.VisualBasic.Specs/UseTestableTimeProvider_specs.vb
Imports NUnit.Framework
Imports CodeAnalysis.TestTools
Imports Qowaiv.CodeAnalysis

Namespace UseTestableTimeProvider_specs

    Public Class Verify

        <Test>
        Public Sub VisualBasic()
            Dim rule As New UseTestableTimeProvider()
            rule.ForVB().
                AddSource("Cases\UseTestableTimeProvider.vb").
                AddReference(Of Qowaiv.Percentage).
                Verify()
        End Sub

    End Class

End Namespace
